### PR TITLE
show spell components in new spell ui and fix color tags in it

### LINKED ISSUE #75669: Magiclysm spell casting menu doesn’t render color tags in the spell description
### Describe the bug

Colors are shown correctly in all other parts of the menu; it is only the description that ignores them.

### Attach save file

n/a

### Steps to reproduce

Cast a spell using the Magiclysm mod.

### Expected behavior

Spell descriptions should not have visible color tags (things like “<color_yellow>” and the like). Those tags should be used to actually change the text color.

### Screenshots

Screenshot 2024-08-14 at 10-50-23 uilist issues from migration #74341 · Issue #75661 · CleverRaven_Cataclysm-DDA

### Versions and configuration

- OS: Linux
    - OS Version: <unknown>
- Game Version: ad2c7edf6a-dirty [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Magiclysm [magiclysm],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth]
]

### Additional context

Note that we cannot actually wrap text into paragraphs properly if it contains color tags. We are currently limited to either drawing a single line of colored text, or wrapping a paragraph, but we cannot do both at the same time. A short–term solution to this problem might simply be to strip the color tags and then render the description using wrapping but no color. At least then the text would be readable and the color tags would not be shown.

### LINKED ISSUE #76138: Spell components not shown in new imgui spell casting ui
### Describe the bug

inside the new spell menu, material spell components (like water for the freshen up spell) are missing from the spell info.

### Attach save file

N/A

### Steps to reproduce

1. create a world with magicalysm
2. select the "Dabbler (Freshen Up)"
3. in game open spell menu and look at the spell info of Freshen Up
4. look of the missing material components you are probably lacking to cast the spell

### Expected behavior

material spell components show in new ui like in the old one:
image

### Screenshots

current ui with the bug:
image

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.22621.4037 (22H2)
- Game Version: 88ecaf6 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth],
    <color_cyan>Arcana and Magic Items</color> [Arcana],
    Aftershock [aftershock],
    Blaze Industries [blazeindustries],
    Magiclysm [magiclysm],
    Mind Over Matter [mindovermatter],
    Xedra Evolved [xedra_evolved],
    Pixel's Various Fuckery [Pix's Shit],
    World on Fire [World_on_fire],
    <color_cyan>Arcana</color>/Aftershock Patchmod [Arcana_aftershock_Patch],
    <color_cyan>Arcana</color>/Magiclysm Patchmod [Arcana_Magiclysm_Patch],
    Mythical Martial Arts [MMA],
    <color_yellow>Mythical Martial Arts - Mods Extension</color> [MMA_Mods_Extension],
    Bombastic Perks [bombastic_perks],
    Blazemod - Vehicle Additions Pack [blazemod],
    Miscellaneous Magiclysm Expansions [demon_spider_a],
    Magiclysm Revamp and Additions [magicrevamp],
    Culinary Days Ahead [culinary_days_ahead]
]

### Additional context

separated this to a new issue from a report in this issue #75669, since I plan to tackle in separately. Have personally confirmed in happening during gameplay.

### PR DESCRIPTION
**Pull request by ThePotatomancer**
fixes color tags in spell info!
Currently the new ui does not display spell
components, and there are colors tags in spell data that are ignored
also changed the message when lacking energy to cast a resource to use the resource name instead of always using stamina :P

#### Summary
Interface "add spell components back to casting menu and fix color tags in it"

#### Purpose of change
resolves #75669 and fixes #76138

#### Describe the solution
use `(cataimgui::draw_colored_text)` in places where we have color tags
change the display of spell components to also use imgui thus show them in spell info again.

#### Describe alternatives you've considered
I considered replacing 'get_folded_components_list' with an approach more oriented for imgui but decided the PR started to creep and it's better to achieve in another PR. 

#### Testing
1. create a world with magicalysm and Mind Over Matter
2. in debug menu add the player a concentration spell and a component spell (I recommend decaying club and spark sight)
3. in game open spell menu and look at the spell info of the previously added spells
4. ensure the materials and spell info appears without color tags

probably worth checking a spell with tool requirements, I'm not knowledgeable enough on what spells have those requirements since they seem quite rare.
cast until out of mana and check the "not enough" message that it refers to mana instead of stamina where appropriate 

#### Additional context

this is the new fixed state: 
image
examples of the broken state are aplenty in the issues referenced
I plan to tackle the popup size issues along with the wrapping issues in uilist in another PR

### CODE REVIEW

File: src/magic.cpp [hunk 1]
```diff
@@ -2742,13 +2742,15 @@ void spellcasting_callback::display_spell_info( size_t index )
         if( !sp.components().get_components().empty() ) {
             for( const std::string &line : sp.components().get_folded_components_list(
                      width - 2, c_light_gray, pc.crafting_inventory( pc.pos(), 0, false ), return_true<item> ) ) {
-                info_txt.emplace_back( line );
+                ImGui::TextWrapped("%s", remove_color_tags(line).c_str());
```
Comment (db48x) [inline src/magic.cpp:2745]: Since you’re no longer using the `info_txt` property, go ahead and remove it from the class as well.

Comment (ThePotatomancer) [reply src/magic.cpp]: removed it along with another unused property 👍

**Comment by ThePotatomancer** [comment]
hopefully I doesn't count as too much feature creep, since the color tags made the component list unreadable and I found that `cataimgui::draw_colored_text` works like a charm to fix it, so I fixed other places in the same menu with that color issue

**Comment by db48x** [comment]
There’s another issue filed somewhere where I discuss all the things we’re doing wrong with text wrapping at the moment. In particular, I’ll be rewriting `draw_colored_text` so that it can do both text wrapping and colored text at the same time (wow!).

**Review by db48x — APPROVED** [review]

**Pull request merged**